### PR TITLE
Bug 1680783: Increase the resolution of first_run_date

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: glean-core/android/src/test/java/mozilla/telemetry/glean/TestUtil.kt [hunk 1]
```diff
@@ -319,3 +319,40 @@ fun <T> capture(value: ArgumentCaptor<T>): T {
...
+fun waitForPingContent(
+    pingName: String,
+    pingReason: String?,
+    server: MockWebServer,
+    maxAttempts: Int = 3
+): JSONObject?
+{
+    var parsedPayload: JSONObject? = null
+    var attempts = 0
+    do {
```
Comment (chutten) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/TestUtil.kt:339]: Is this loop different from `for (attempts in 1..maxAttempts)`?

File: glean-core/android/src/test/java/mozilla/telemetry/glean/scheduler/MetricsPingSchedulerTest.kt [hunk 2]
```diff
@@ -261,6 +261,16 @@ class MetricsPingSchedulerTest {
             serverEndpoint = "http://" + server.hostName + ":" + server.port
         ))
 
+        // Catch the automatics metrics ping
```
Comment (chutten) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/scheduler/MetricsPingSchedulerTest.kt:264]: "automatics"?

File: glean-core/src/lib.rs [hunk 3]
```diff
@@ -445,14 +446,16 @@ impl Glean {
         // so that it can't be accessed until this function is done.
         let _lock = self.upload_manager.clear_ping_queue();
 
-        // There is only one metric that we want to survive after clearing all
-        // metrics: first_run_date. Here, we store its value so we can restore
-        // it after clearing the metrics.
+        // There is only two metrics that we want to survive after clearing all
+        // metrics: first_run_date and first_run_hour. Here, we store its value
+        // so we can restore it after clearing the metrics.
```
Comment (chutten) [inline glean-core/src/lib.rs:451]:
```suggestion
        // There are only two metrics that we want to survive after clearing all
        // metrics: first_run_date and first_run_hour. Here, we store their values
        // so we can restore them after clearing the metrics.
```

**Review by chutten — APPROVED** [review]

File: glean-core/android/src/test/java/mozilla/telemetry/glean/TestUtil.kt (hunk 1 above)
Comment (mdboom) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/TestUtil.kt:339]: No, good point.  I actually copied this from the [test utilities for the sample app...](https://github.com/mozilla/glean/blob/main/samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/SharedTestUtils.kt#L65)

File: glean-core/android/src/test/java/mozilla/telemetry/glean/scheduler/MetricsPingSchedulerTest.kt (hunk 2 above)
Comment (mdboom) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/scheduler/MetricsPingSchedulerTest.kt:264]: lol.  "automatic"

**Pull request merged**